### PR TITLE
refactor: Move admin comments into drawer

### PR DESCRIPTION
**Pull request by ChangoMan**
## Description

I'm thinking about moving the comments into a `drawer`, instead of using the `collapse`, so there's less animation and jank when adding or viewing the comments. Let me know how you all feel about this approach!

### Screenshot

Screenshot 2024-08-24 at 11 07 14 AM

### CODE REVIEW

File: packages/nextjs/app/admin/_components/SubmissionComments.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+                disabled={isPending}
+                onChange={field => {
+                  setNewComment(field.target.value);
+                }}
+              />
+            </label>
+            <button className="mt-4 btn btn-sm btn-accent" disabled={isPending}>
+              {isPending && <span className="loading loading-spinner w-4 h-4"></span>}
+              Add Comment
+            </button>
```
Comment (ChangoMan) [inline packages/nextjs/app/admin/_components/SubmissionComments.tsx:109]: Also adding a loading state to the add comment form

**Comment by carletex** [comment]
I like it!

Could you resolve conflicts? (`getFormattedDateTime` doesn't exist anymore in the file since you move it in the inner component, but Damu uses it on #61)

**Comment by ChangoMan** [comment]
@carletex Merge conflicts should be resolved!

**Review by carletex — APPROVED** [review]
Working great @ChangoMan thanks!!

Just pushed https://github.com/BuidlGuidl/extensions-hackathon/pull/64/commits/718d8d132019ecf42241d3b47da91189e97db4e9. Felt a bit off to have the `getFormattedDateTime` util inside the SubmissionComments component.

**Comment by carletex** [comment]
(nitpick for future PR: even if you can close the drawer by clicking outside, maybe we could also have a `X` on the top right of the drawer) @ChangoMan

**Pull request merged**